sirepo/package_data/static/js/raydata.js

```diff
@@ -725,12 +702,50 @@ SIREPO.app.directive('scanSelector', function() {
                 );
             };
 
+            $scope.setColumnHeaders = function() {
+            $scope.columnHeaders = [
+                'selected',
+                ...$scope.defaultColumns,
+                ...appState.models.metadataColumns.selected
+            ];
+            };
+
+            $scope.setAvailableColumns = function() {
+                $scope.availableColumns = masterListColumns.filter((value) => {
+                    return value !== 'uid' && ! $scope.columnHeaders.includes(value);
+                });
+            };
+
+            $scope.showDeleteButton = (index) => {
+                return index > $scope.defaultColumns.length && index === hoveredIndex;
+            };
+
             $scope.showSearchButton = function() {
                 return $scope.searchForm.$dirty && $scope.searchStartTime && $scope.searchStopTime;
             };
 
+            $scope.setColumnHeaders();
             $scope.toggleScanSelection = raydataService.maybeToggleScanSelection;
             appState.whenModelsLoaded($scope, () => $scope.search());
+
+            requestSender.sendStatelessCompute(
+                appState,
+                (json) => {
+                    masterListColumns = json.columns;
+                    $scope.setAvailableColumns();
+                },
+                {
+                    method: 'get_scan_fields',
+                    scans: []
```

Comment:
I don't believe this is needed.

Reply:
bd781ab41c48e24e796b4abc76b8e738de72ba2d